File: 2BIT/letní/IPP/Projekt 2/ippcode18_error.py
"""
Rozhrani pro chyby a navratove kody
@project: IPP18 proj2
@brief: Rozhrani pro chyby a navratove kody
@author: Matej Mitas
@file: ippcode18_error.py
"""

import sys

class Error:

	err_codes = {
		# parametry a soubory
		'10': 'Spatne pouziti parametru spousteni',
		'11': 'Nevalidni vstupni soubor',
		'12': 'Nevalidni vystupni soubor',
		# lea a sya
		'31': 'Co jsi mi to poslal za XML?',
		'32': 'Lexikalni nebo Syntakticka chyba',
		# interpretace
		'52': 'Semanticka chyba',
		'53': 'Spatne typy operandu',
		'54': 'Promenna neexistuje (ramec OK)',
		'55': 'Ramec neexistuje',
		'56': 'Chybejici hodnota',
		'57': 'Deleni nulou',
		'58': 'Chybna prace s retezcem',
		# vnitrni chyba, jake je vyvolani?
		'99': 'Vnitrni chyba interpretu',
	}

	"""
	Konstruktor
	@param kod
	@param vlastni zprava pro uzivatele pri chybe
	@return None
    """
	def __init__(self, code, custom_message = ''):
		ec = str(code)
		if ec in Error.err_codes:
			sys.stderr.write("▷ {}\n⎆ {}\n".format(Error.err_codes[ec], custom_message))
			sys.exit(code)

File: 2BIT/letní/IPP/Projekt 2/_ippcode18_error.php
<?php

/*
Rozhrani pro chyby
@project: IPP18 proj2
@brief: Rozhrani pro chyby
@author: Matej Mitas
@file: ippcode18_error.php
*/

class ErrorCode {
	
	/**
     * Pole pro vysvetleni k chybovym kodum
     * @var pole
     */
	private $err_codes = [
		'10' => ">Spatne pouziti parametru spousteni\nNavratovy kod: 10",
		'11' => ">Nepodarilo se otevrit vstupni soubor\nNavratovy kod: 11",
		'12' => ">Nepodarilo se otevrit vystupni soubor\nNavratovy kod: 12",
		'21' => ">Lexikalni chyba analyzatoru\nNavratovy kod: 21",
		'22' => ">Hlavni souboru musi vzdy zacinat '.IPPcode18'"
	];

	/**
	 * Vypiseme chybovou zpravu a ukoncime program s kodem
	 * @return void
	 */
	public function yield_err() {
		fwrite(STDERR, $this->err_codes[$this->code]);
		exit($this->code);
	}

	/**
	 * Konstruktor
	 * @param chybovy kod
	 * @return void
	 */
	public function __construct($code) {
		$this->code = $code;
		$this->yield_err();
	}
}

?>